core/workspace_svg.js

```diff
@@ -66,6 +69,13 @@ Blockly.WorkspaceSvg = function(options, opt_blockDragSurface) {
     this.blockDragSurface_ = opt_blockDragSurface;
   }
 
+  if (opt_wsDragSurface) {
+    this.workspaceDragSurface_ = opt_wsDragSurface;
+  }
+
+  this.useWorkspaceDragSurface_ =
```

Comment:
We could do the is3dSupported check in inject.js, then just check whether a drag surface has been passed in here.